### ISSUE TITLE
`format` and `clean-ns` disagree for indentation level when using `:ns-import-classes-indentation :same-line`

### ISSUE THREAD

**Issue by vemv** [clojure-lsp/clojure-lsp#1421]
Hi!

I'm using `com.github.clojure-lsp/clojure-lsp-standalone {:mvn/version "2022.12.09-15.51.10"}` to format and check namespaces' formatting. This is a CI-oriented check.

I took https://github.com/clojure-lsp/lein-clojure-lsp as inspiration. So I run the `format` and `clean-ns` ops, either in plain mode or `dry` mode.

The problem I'm experiencing is that `format` and `clean-ns` can produce slightly different output. So if I apply `format`'s formatting, I will make `clean-ns --dry` upset, and vice versa.

Which makes it impossible to run both `format --dry` and `clean-ns --dry` as a single CI check.

This is the point of contention:

```diff
(ns foo
  (:import [java.nio.file FileSystem
-           FileSystems
-           Path
-           PathMatcher]))
+                         FileSystems
+                         Path
+                         PathMatcher]))

```

i.e. `format` will produce the red version, and `clean-ns`, the green version.

This is our .lsp config:

```clj
:clean {:ns-inner-blocks-indentation :same-line
        :ns-import-classes-indentation :same-line}
```

...and .cljfmt.edn:

```clj
{:remove-surrounding-whitespace?  true
 :remove-trailing-whitespace?     true
 :remove-consecutive-blank-lines? true
 :insert-missing-whitespace?      true
 :align-associative?              false
```

Does this look like a bug?

Thanks - V

**Comment by ericdallo** [comment]
At Nubank we use `:ns-inner-blocks-indentation :same-line` and format + clean-ns, so I think this bug maybe happens for specific cases, maybe only for `:import` and not `:require`?

**Comment by ericdallo** [comment]
Even so, it seems to me the clean-ns is the correct one for your case right? if so, we should check how to fix on cljfmt or create a config to allow that there

**Comment by ericdallo** [comment]
Actually for me, cleaning:
```clojure
  (:import [java.nio.file FileSystem
            FileSystems
            Path
            PathMatcher])
```
results in:
```clojure
  (:import [java.nio.file
            FileSystem
            FileSystems
            Path
            PathMatcher])
```

**Comment by ericdallo** [comment]
> Perhaps a simple approach (simpler than teaching cljfmt something that doesn't know how to do currently) would be to ensure that that the each :import clause is expressed in one line, no matter its length.

We already support that via `:clean :sort :import-classes :classes-per-line` setting with value `-1` as mentioned [here](https://github.com/clojure-lsp/clojure-lsp/blob/master/docs/all-available-settings.edn#L20)

>I can create a repro if needed, as it's not impossible that this codebase (new for me) has something odd hidden. But hopefully it's all simple and accurately captured in my OP.

yeah, would be nice to understand when it happens so we can avoid that inconsistency

**Comment by ericdallo** [comment]
I'll check the sample and improve the docs to mention the other section

**Comment by ericdallo** [comment]
Right, testing your repro the issue seems to be on `:ns-import-classes-indentation :same-line`

**Comment by ericdallo** [comment]
So, I'm still not sure there is something to improve on lsp side, when `:ns-import-classes-indentation` is `:same-line` we do this:
```clojure
(ns licenses
  (:import
   [java.nio.file FileSystem
                  FileSystems
                  Path
                  PathMatcher]))
```
But cljfmt do this:
```clojure
(ns licenses
  (:import
   [java.nio.file FileSystem
    FileSystems
    Path
    PathMatcher]))
```

And that looks pretty wrong to me as one class is in the same line (`FileSystem`) and others on the next line, it doesn't seem to me that is a standard, never saw that kind of formatting, is there any place mentioning that kind of sorting?

**Comment by ericdallo** [comment]
I understand your point, but TBH all those options are already a lot, and I'd like to avoid if possible adding yet another option to support the 'uglier' style if no users (even you, as will use the -1 value of that setting) have interest in using it, just to match other toolings that only do that for simplicity and not for the real semantic of that :/

**Comment by ericdallo** [comment]
It doesn't look to me too composable :/ as we have other tasks like `diagnostics` and may have more in the future.
what we could do maybe in the future is call cljfmt to format the ns always after cleaning, his way we would have always both behaving the same
